test/e2e/serial/workload_placement_by_tm_policy_test.go

```diff
@@ -0,0 +1,347 @@
+			nrtCandidates := e2enrt.FilterZoneCountEqual(nrts, 2)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with 2 NUMA Zones: found %d", len(nrtCandidates)))
+			}
+			By("filtering available nodes with allocatable resources on each NUMA zone that can match request")
+			nrtCandidates = e2enrt.FilterAnyZoneMatchingResources(nrtCandidates, requiredRes)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with NUMA zones each of them can match requests: found %d", len(nrtCandidates)))
+			}
+
+			candidateNodeNames := e2enrt.AccumulateNames(nrtCandidates)
+			// nodes we have now are all equal for our purposes. Pick one at random
+			// TODO: make sure we can control this randomness using ginkgo seed or any other way
+			targetNodeName, ok := candidateNodeNames.PopAny()
+			Expect(ok).To(BeTrue(), "cannot select a target node among %#v", candidateNodeNames.List())
+			unsuitableNodeNames := candidateNodeNames.List()
+
+			By(fmt.Sprintf("selecting target node %q and unsuitable nodes %#v (random pick)", targetNodeName, unsuitableNodeNames))
+
+			padInfo := paddingInfo{
+				pod:                 pod,
+				targetNodeName:      targetNodeName,
+				unsuitableNodeNames: unsuitableNodeNames,
+				unsuitableFreeRes:   unsuitableFreeRes,
+			}
+
+			By("Padding nodes to create the test workload scenario")
+			paddingPods := setupPadding(fxt, nrtList, padInfo)
+
+			By("waiting for ALL padding pods to go running - or fail")
+			failedPods := e2ewait.ForPodListAllRunning(fxt.Client, paddingPods)
+			for _, failedPod := range failedPods {
+				// ignore errors intentionally
+				_ = objects.LogEventsForPod(fxt.K8sClient, failedPod.Namespace, failedPod.Name)
+			}
+			Expect(failedPods).To(BeEmpty())
+
+			// TODO: smarter cooldown
+			time.Sleep(18 * time.Second)
+			for _, unsuitableNodeName := range unsuitableNodeNames {
+				dumpNRTForNode(fxt.Client, unsuitableNodeName, "unsuitable")
+			}
+			dumpNRTForNode(fxt.Client, targetNodeName, "target")
+
+			By("checking the resource allocation as the test starts")
+			nrtListInitial, err := e2enrt.GetUpdated(fxt.Client, nrtList, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("running the test pod")
+			dumpPod(pod)
+
+			By("running the test pod")
+			err = fxt.Client.Create(context.TODO(), pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting for the pod to be scheduled")
+			updatedPod, err := e2ewait.ForPodPhase(fxt.Client, pod.Namespace, pod.Name, corev1.PodRunning, 2*time.Minute)
+			if err != nil {
+				_ = objects.LogEventsForPod(fxt.K8sClient, pod.Namespace, pod.Name)
+			}
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod.Spec.NodeName, targetNodeName))
+			Expect(updatedPod.Spec.NodeName).To(Equal(targetNodeName),
+				"node landed on %q instead of on %v", updatedPod.Spec.NodeName, targetNodeName)
```

Comment:
> Expect(updatedPod.Spec.NodeName).To(Equal(targetNodeName),
				"node landed on %q instead of on %v", updatedPod.Spec.NodeName, targetNodeName)

"pod landed on .."

Reply:
thanks, fixed